### ISSUE TITLE
Inaccurate comment for AES_CTR_xcrypt_buffer()?

### ISSUE THREAD

**Issue by amosnier** [kokke/tiny-AES-c#77]
Hi,

Correct me if I am wrong, but I believe the comment "buffer size MUST be multiple of AES_BLOCKLEN" is inaccurate for AES_CTR_xcrypt_buffer() (in aes.h).

What if it's not? I believe Wikipedia's description of the case:

- The last partial block of plaintext is XORed with the first few bytes of the last keystream block, producing a final ciphertext block that is the same size as the final partial plaintext block.

would apply to this implementation.

Requiring padding for CTR would also, in my interpretation, fail to fulfill NIST SP 800-38A, that specifies padding for ECB, CBC, and CFB modes only.

In other words, I believe this implementation is better than advertised, and the comment should be updated to reflect that. :-)

Thanks for putting this implementation in the public domain.

Best regards,

Alain Mosnier

**Comment by kokke** [comment]
Hi @amosnier and thanks for taking an interest in this project :)

You are 100% right: That comment is not appropriate for CTR-mode.

The API was changed to be thread-safe a few days ago in #76 where the comment snuck in.

I'll remove the comment right away :)

Thanks for spotting it!
